### ISSUE TITLE
Axe issue: "All page content must be contained by landmarks"

### ISSUE THREAD

**Issue by colinrotherham** [alphagov/govuk-frontend#1604]
Hello 👋 

We’ve noticed for a while that prototype kit and GOV.UK Frontend examples keep the “Back link” and “Phase banner” outside landmark regions.

i.e. They’re not in a `<header>`, `<footer>` or `<main>`.

This throws up moderate issues in axe:
https://dequeuniversity.com/rules/axe/3.3/region?application=AxeChrome

### Components not in `<main>` or `<header>`
This example shows showing both components outside regions:  
http://govuk-frontend-review.herokuapp.com/full-page-examples/upload-your-photo

![Components outside  main](https://user-images.githubusercontent.com/415517/66395735-18cbf580-e9d0-11e9-978b-cdd0ca4fa496.png)

### But the "Breadcrumbs" component _is_ in `<main>`
This example shows breadcrumbs inside a region:  
http://govuk-frontend-review.herokuapp.com/full-page-examples/renew-driving-licence

![Breadcrumbs component inside  main](https://user-images.githubusercontent.com/415517/66395830-4dd84800-e9d0-11e9-889c-8eb3ab62f0d9.png)

Here's what we see in the axe sidebar when follow the examples (with “Back link” and “Phase banner” outside landmark regions):

![Axe report](https://user-images.githubusercontent.com/415517/66396125-f686a780-e9d0-11e9-81b3-fb411141704c.png)

It would be great to see the research behind when to use a landmark or not.

Should this be logged under https://github.com/alphagov/govuk-frontend/issues/1280#issuecomment-509588851?

Thanks

**Comment by nickcolley** [comment]
Just to clarify, the 'govuk-frontend-review' application is not intended to be guidance and contains intentionally legacy and therefore out of date patterns for test the 'compatibility mode'.

So that's the reason why one of the examples in there is wrong.

In the official guidance on the Design System website we intentionally do not have the phase banner, back links, or breadcrumbs in the 'main' element.

This is for two reasons I know of:
- the skip link is less useful if it does not skip the phase banner, back links and breadcrumbs
- too many navigation landmarks on GOV.UK have been reported to be an issue so we intentionally have removed some navigation landmarks in order to make it clear what the main navigation is.

I don't have the original research for the latter decision, but will ask around and see if I can dig it up.

Will leave this in triage for us to decide as a team what direction we want to take, thanks a lot for raising this I'm sure others will run into this too.

**Comment by nickcolley** [comment]
With the help from @36degrees @selfthinker I've pulled out some of the historical stuff:

> It turns out some of our nav tags were redundant. They either added no extra meaning to the links they wrapped or were unnecessary as user agents got the same information from the links' context instead.
>
> Among those that were needed, the navigational theme that grouped their contents was sometimes unclear.
> 
> The above came out of an accessibility review by Léonie Watson. In it she explained that having too many nav tags devalued their use for screen reader users. It was suggested that when multiple ones were necessary we could help those users find the links they needed by identifying them properly.

- [Rethinking navigation by Tom Byers](https://insidegovuk.blog.gov.uk/2013/07/03/rethinking-navigation/)
- [GitHub change to GOV.UK](https://github.com/alphagov/static/pull/238)

**Comment by colinrotherham** [comment]
@nickcolley This is exactly the sort of thing I was after! Thank you for the detective work 🕵

**Comment by hannalaakso** [comment]
I've added this to https://github.com/alphagov/govuk-frontend/issues/1280#issuecomment-509588851 as it can also seen on the GOV.UK Frontend Template. Thanks again for raising this @colinrotherham

**Comment by nickcolley** [comment]
Looks like this has been resolved and added to documentation for people to find in the future so will close this one out, thanks :)

**Comment by 36degrees** [comment]
Re-opening this in favour of #1949 as this has more context.

Whilst we previously decided against moving the phase banner, back link or breadcrumbs into the `<main>` element, we have recently discussed whether it would make sense instead to extend the `<header>` to include them alongside the header component.

**Comment by 36degrees** [comment]
We could also revisit wrapping the breadcrumbs and back link in navigation landmarks, taking into consideration https://insidegovuk.blog.gov.uk/2013/07/03/rethinking-navigation/ and validating with user research.

**Comment by robertparkinson** [comment]
I have come across this issue recently in our code and have worked up a potentional solution for the phase banner being moved into the header. How do I go about getting rights to push a PR for discussion?

**Comment by 36degrees** [comment]
@robertparkinson we do not grant write permissions to users outside of the team, but we're more than happy for you to [raise a pull request from a fork](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork).

It does sound like the change you're suggesting would be fairly substantial (and breaking), so it would need to be managed carefully. To avoid any wasted effort on your part, are you able to describe the change that you're planning on proposing first?

**Comment by robertparkinson** [comment]
Thanks @36degrees I've opened a PR. Would love your feedback and have just focussed on the phase banner as that should be an easier conversation than surroundign the back button with nav tags. https://github.com/alphagov/govuk-frontend/pull/1984

**Comment by matthewmascord** [comment]
Having the ability to add additional content before the HEADER closing tag would definitely be a step forward. This would be for content that is semantically part of the header for screen reader purposes, for example, a departmental banner, but visually under the grey bar.

Allowing this would mean screenreader users can get to the main content quicker and avoid Aria difficulties around having multiple banner landmarks on the same page.

**Comment by jorgeponto** [comment]
> With the help from @36degrees @selfthinker I've pulled out some of the historical stuff:
> 
> > It turns out some of our nav tags were redundant. They either added no extra meaning to the links they wrapped or were unnecessary as user agents got the same information from the links' context instead.
> > Among those that were needed, the navigational theme that grouped their contents was sometimes unclear.
> > The above came out of an accessibility review by Léonie Watson. In it she explained that having too many nav tags devalued their use for screen reader users. It was suggested that when multiple ones were necessary we could help those users find the links they needed by identifying them properly.
> 
> * [Rethinking navigation by Tom Byers](https://insidegovuk.blog.gov.uk/2013/07/03/rethinking-navigation/)
> * [GitHub change to GOV.UK](https://github.com/alphagov/static/pull/238)

Yes. It Is understandable the rationale of Leónie Watson, but a div title="breadcrumb" and a aria-current="page" at the last link (if I'm here why I need a link?) of the sequence is a good improve to screenreader users understand the structure of this element.

**Comment by brunswickcomputing** [comment]
skip-link still throws the same warning in Axe. i can't see any reference to this being tackled so far. any plans or have I missed something?

**Comment by 36degrees** [comment]
We should consider the skip link component as part of this issue, but worth noting that [Deque themselves say](https://dequeuniversity.com/rules/axe/4.0/region):

> It is best practice to contain all content **excepting skip links**, within distinct regions such as the header, nav, main, and footer.

**Comment by owenatgov** [comment]
We revisited this very briefly in a discussion amongst some of the devs today. The takeaways:

- Whilst this is something worth fixing for the sake of suppressing warnings and saving dev teams time, we are skeptical that it's an actual issue for users so it is still low priority
- We're going to run this past our accessibility specialist to verify that skepticism
- This could be solved by a long term idea to not couple the header component to the `header` landmark which itself could potentially go into v5

**Comment by selfthinker** [comment]
It's worth noting that Axe changed this from a "must" and flagging it as a WCAG fail to a "should" and now flags it as "best practice". See more on [what Deque say about the issue from Axe](https://dequeuniversity.com/rules/axe/4.10/region?application=AxeChrome).

**Comment by selfthinker** [comment]
We have [investigated this issue](https://github.com/alphagov/govuk-frontend/issues/6460) and have decided to:

* move the phase banner into the header element (while also decoupling the header element from the header component) for v6
* not change any code about the back link but add information to the back link page to make people aware of this potential issue and ask for feedback if anyone came across any issues with actual users around this
* (the [breadcrumbs were changed last year](https://github.com/alphagov/govuk-frontend/pull/4995) to be in a navigation landmark)

**Comment by selfthinker** [comment]
I have just checked this in a couple of other accessibility checkers and found that this also gets raised by the Siteimprove checker (under "WAI-ARIA authoring practices") as well as the IBM Equal Access Accessibility Checker (as a violation of 1.3.1 and 2.4.1).

**Comment by selfthinker** [comment]
We will add a link to this issue from the back link guidance. To avoid users needing to look at another page, here are the reasons (from the decision issue linked above) for not changing where the back link is for now and why changing it might be a worse experience:

> None of the options where to put the back link are great:
> * Moving it to the `header` is out of the question as it is more specific to the page you are on.
> * Wrapping it in an `aside` element is not great as it doesn't fit what the spec says.
> * Moving it to the `main` element would cause a WCAG fail of 2.4.1 Bypass Blocks.
> * Keeping it as it is (outside of any landmarks) will still cause an Axe best practice issue and therefore more support requests.
> * Wrapping it in a `nav` element means having a whole landmark for one single link which seems excessive. It is bad practice to have too many landmarks, although if a service has breadcrumbs on other pages this would at least not increase the general amount of landmarks per page (because breadcrumbs and back links are never together on the same page). It is possible that this might make it worse for screen reader users than not having this in a landmark.

Also, more background info on landmarks:

> * Theory of landmarks and practice of landmarks are not fully consistent with each other. As always, what the spec says and how vendors implement it and how users use it can be different.
> * If screen readers care about landmarks, they mostly care about the big ones: main (main), banner (header), contentinfo (footer) and navigation (nav), and out of those mostly about main. (An example of evidence of that is in this [discussion thread](https://webaim.org/discussion/mail_thread?thread=10090).)
> * Screen reader users still mostly use headings to navigate through a page. From the [WebAIM screen reader survey](https://webaim.org/projects/screenreadersurvey10/#landmarks): "While 31.7% of respondents indicate that they always or often use landmarks when they are present, only 3.7% use this as a primary method for finding information on a lengthy web page."
> * I couldn't find anything reliable that says where little snippets like the back link should go. The question is not just where it makes semantic sense, but much more where screen reader users expect to find it. There is no clear answer for that.
